File: python-elasticemail-examples/examples/with_attachments.py
"""
Email with Attachment Example

Usage: python examples/with_attachments.py
"""

import base64
import datetime
import os
import sys

import ElasticEmail

sys.path.insert(0, os.path.dirname(__file__))
from ee import FROM, TO, get_configuration, print_api_error


def main():
    configuration = get_configuration()

    file_content = (
        "Sample Attachment\n"
        "==================\n\n"
        "This file was attached to your email.\n"
        "Sent at: {}\n".format(datetime.datetime.utcnow().isoformat())
    )
    encoded = base64.b64encode(file_content.encode()).decode()

    with ElasticEmail.ApiClient(configuration) as api_client:
        emails_api = ElasticEmail.EmailsApi(api_client)

        message = ElasticEmail.EmailTransactionalMessageData(
            Recipients=ElasticEmail.TransactionalRecipient(To=[TO]),
            Content=ElasticEmail.EmailContent(
                From=FROM,
                Subject="Email with Attachment",
                Body=[
                    ElasticEmail.BodyPart(
                        ContentType="HTML",
                        Content="<h1>Your attachment is ready</h1><p>Please find the file attached to this email.</p>",
                    ),
                ],
                # BinaryContent is base64. Total message size limit applies (see account limits).
                Attachments=[
                    ElasticEmail.MessageAttachment(
                        BinaryContent=encoded,
                        Name="sample.txt",
                        ContentType="text/plain",
                    ),
                ],
            ),
        )

        try:
            result = emails_api.emails_transactional_post(message)
            print("Email with attachment sent successfully!")
            print("Transaction ID:", result.transaction_id)
            print("Message ID:", result.message_id)
        except ElasticEmail.ApiException as e:
            print_api_error("send email", e)
            sys.exit(1)


if __name__ == "__main__":
    main()

File: ruby-elasticemail-examples/examples/with_attachments.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

##
# Email with Attachments Example
#
# Usage: ruby examples/with_attachments.rb

require_relative "ee"
require "base64"

emails_api = ElasticEmail::EmailsApi.new

file_content = <<~TEXT
  Sample Attachment
  ==================

  This file was attached to your email.
  Sent at: #{Time.now.utc.iso8601}
TEXT
encoded = Base64.strict_encode64(file_content)

begin
  result = emails_api.emails_transactional_post(
    ElasticEmail::EmailTransactionalMessageData.new(
      recipients: ElasticEmail::TransactionalRecipient.new(to: [TO]),
      content: ElasticEmail::EmailContent.new(
        from: FROM,
        subject: "Email with Attachment",
        body: [
          ElasticEmail::BodyPart.new(
            content_type: "HTML",
            content: "<h1>Your attachment is ready</h1><p>Please find the file attached to this email.</p>"
          )
        ],
        # BinaryContent is base64. Total message size limit applies (see account limits).
        attachments: [
          ElasticEmail::MessageAttachment.new(
            binary_content: encoded,
            name: "sample.txt",
            content_type: "text/plain"
          )
        ]
      )
    )
  )

  puts "Email with attachment sent successfully!"
  puts "Transaction ID: #{result.transaction_id}"
  puts "Message ID: #{result.message_id}"
rescue ElasticEmail::ApiError => e
  api_error("send email", e)
end
